src/citrine/gemtables/variables.py

```diff
@@ -134,9 +139,10 @@ class AttributeByTemplate(Serializable['AttributeByTemplate'], Variable):
         a short human-readable name to use when referencing the variable
     headers: list[str]
         sequence of column headers
-    template: LinkByUID
+    template: Union[UUID, str, LinkByUID, AttributeTemplate]
         attribute template that identifies the attribute to assign to the variable
-    attribute_constraints: list[list[LinkByUID, Bounds]]
+    attribute_constraints: list[list[Union[UUID, str, LinkByUID, AttributeTemplate], Bounds]]
```

Comment:
If you defined the new `template_type` in the other file, it seems like it would be worth doing here too